src/renderer/qssWidget/js/qssCustomStepperBrightnessWidget.js

```diff
@@ -0,0 +1,342 @@
+                type: "gpii.qssWidget.button",
+                container: "{that}.dom.decButton",
+                options: {
+                    model: {
+                        label: "{stepperBrightness}.model.messages.decrementButton"
+                    },
+                    invokers: {
+                        activate: "{stepperBrightness}.activateDecBtn"
+                    }
+                }
+            },
+            // similar to qssWidget component
+            windowKeyListener: {
+                type: "fluid.component",
+                options: {
+                    gradeNames: "gpii.app.keyListener",
+                    target: {
+                        expander: {
+                            funcName: "jQuery",
+                            args: [window]
+                        }
+                    },
+                    events: {
+                        onSubtractPressed: null,
+                        onEqualsPressed:   null,
+                        onAddPressed:      null
+                    },
+                    listeners: {
+                        onAddPressed:      "{stepperBrightness}.activateIncBtn",
+                        onEqualsPressed:   "{stepperBrightness}.activateIncBtn",
+                        onSubtractPressed: "{stepperBrightness}.activateDecBtn"
+                    }
+                }
+            }
+        }
+    });
+
+    /**
+     * Invoked whenever the value of the setting has reached its upper or lower
+     * bound and an attempt is made to go beyond that bound. In that case an
+     * error tone is played. If at least `specialErrorBoundHitTries` number of
+     * times this has happened, in addition to the error tone, a notification
+     * is shown to the user.
+     * @param {Component} that - The `gpii.qssWidget.stepperBrightness` instance.
+     * @param {String} errorMessage - The message to be displayed in the QSS
+     * notification.
+     */
+    gpii.qssWidget.stepperBrightness.handleBoundReached = function (that, errorMessage) {
+        gpii.psp.playSound(that.options.sounds.boundReached);
+
+        if (that.boundReachedHits >= that.options.specialErrorBoundHitTries) {
+            that.events.onNotificationRequired.fire({
+                description: errorMessage,
+                closeOnBlur: true
+            });
+        }
+    };
+
+    /**
+     * Invoked whenever the increment button is activated. Takes care of
+     * increasing the setting's value with the amount specified in the setting's
+     * schema, animating the button appropriately and/or firing an event if
+     * an attempt is made to increase value above the maximum allowed value.
+     * @param {Component} that - The `gpii.qssWidget.stepperBrightness` instance.
+     * @param {jQuery} button - The jQuery object repesenting the increment button
+     * in the QSS stepper widget.
+     */
+    gpii.qssWidget.stepperBrightness.activateIncButton = function (that, button) {
```

Comment:
This implementation is essentially identical with activateDecButton - please consolidate

Reply:
Done